File: csharp/Svix/ApiInternal/Autoconfig.cs
// this file is @generated
#nullable enable
using Microsoft.Extensions.Logging;
using Svix.Models;

namespace Svix.ApiInternal
{
    public class Autoconfig(SvixClient client)
    {
        readonly SvixClient _client = client;

        public AutoconfigDestination Destination
        {
            get => new AutoconfigDestination(_client);
        }

        public AutoconfigEndpoint Endpoint
        {
            get => new AutoconfigEndpoint(_client);
        }

        /// <summary>
        /// Get an AutoConfig subscription, including the bound endpoint or destination if any.
        /// </summary>
        public async Task<AutoConfigSubscriptionOut> GetAsync(
            string appId,
            string autoconfigId,
            CancellationToken cancellationToken = default
        )
        {
            try
            {
                var response =
                    await _client.SvixHttpClient.SendRequestAsync<AutoConfigSubscriptionOut>(
                        method: HttpMethod.Get,
                        path: "/api/v1/app/{app_id}/autoconfig/{autoconfig_id}",
                        pathParams: new Dictionary<string, string>
                        {
                            { "app_id", appId },
                            { "autoconfig_id", autoconfigId },
                        },
                        cancellationToken: cancellationToken
                    );
                return response.Data;
            }
            catch (ApiException e)
            {
                _client.Logger?.LogError(e, $"{nameof(GetAsync)} failed");

                throw;
            }
        }

        /// <summary>
        /// Get an AutoConfig subscription, including the bound endpoint or destination if any.
        /// </summary>
        public AutoConfigSubscriptionOut Get(string appId, string autoconfigId)
        {
            try
            {
                var response = _client.SvixHttpClient.SendRequest<AutoConfigSubscriptionOut>(
                    method: HttpMethod.Get,
                    path: "/api/v1/app/{app_id}/autoconfig/{autoconfig_id}",
                    pathParams: new Dictionary<string, string>
                    {
                        { "app_id", appId },
                        { "autoconfig_id", autoconfigId },
                    }
                );
                return response.Data;
            }
            catch (ApiException e)
            {
                _client.Logger?.LogError(e, $"{nameof(Get)} failed");

                throw;
            }
        }
    }
}

File: python/svix/api_internal/autoconfig.py
# This file is @generated
from ..models import AutoConfigSubscriptionOut
from .autoconfig_destination import (
    AutoconfigDestination,
    AutoconfigDestinationAsync,
)
from .autoconfig_endpoint import (
    AutoconfigEndpoint,
    AutoconfigEndpointAsync,
)
from .common import ApiBaseAsync, ApiBaseSync


class AutoconfigAsync(ApiBaseAsync):
    @property
    def destination(self) -> AutoconfigDestinationAsync:
        return AutoconfigDestinationAsync(self._client, self._httpx_client)

    @property
    def endpoint(self) -> AutoconfigEndpointAsync:
        return AutoconfigEndpointAsync(self._client, self._httpx_client)

    async def get(self, app_id: str, autoconfig_id: str) -> AutoConfigSubscriptionOut:
        """Get an AutoConfig subscription, including the bound endpoint or destination if any."""
        response = await self._request_asyncio(
            method="get",
            path="/api/v1/app/{app_id}/autoconfig/{autoconfig_id}",
            path_params={
                "app_id": app_id,
                "autoconfig_id": autoconfig_id,
            },
        )
        return AutoConfigSubscriptionOut.model_validate(response.json())


class Autoconfig(ApiBaseSync):
    @property
    def destination(self) -> AutoconfigDestination:
        return AutoconfigDestination(self._client, self._httpx_client)

    @property
    def endpoint(self) -> AutoconfigEndpoint:
        return AutoconfigEndpoint(self._client, self._httpx_client)

    def get(self, app_id: str, autoconfig_id: str) -> AutoConfigSubscriptionOut:
        """Get an AutoConfig subscription, including the bound endpoint or destination if any."""
        response = self._request_sync(
            method="get",
            path="/api/v1/app/{app_id}/autoconfig/{autoconfig_id}",
            path_params={
                "app_id": app_id,
                "autoconfig_id": autoconfig_id,
            },
        )
        return AutoConfigSubscriptionOut.model_validate(response.json())
